### ISSUE TITLE
Condition WHERE IN not working

### ISSUE THREAD

**Issue by comm1x** [vsch/kotlin-jdbc#17]
When I try to bind list of params for `IN` condition, I get 0 count of rows.

This code fails, when country with `id=1` exists.

```
val q = sqlQuery("SELECT * FROM country WHERE id IN (:ids)")
            .inParams("ids" to listOf(1))

assertEquals(1, session(dataSource).count(q))
```

I also try `mapOf("ids" to listOf(1))` but it also not working.
Db: mysql-5.7
Kotlin-jdbc version 0.5.0

**Comment by comm1x** [comment]
I am not sure, but seems the problem is in

Helpers.kt:30
```
setParam(idx, param.value)
```
param.value is array, and it leading the error. Using `value` instead may fix it

**Comment by vsch** [comment]
@comm1x, the issue is in SqlQueryBase.populateParams. The section handling lists calls `stm.setTypedParam(index + idx+1, paramItem, param)` in error. The `param` argument is a collection, so set param passes it as Objet.

Instead, the code for handing the items of a collection should be:

```
                        when (paramItem) {
                            is Parameter<*> -> stmt.setTypedParam(it + idx + 1, param)
                            else -> stmt.setParam(it + idx + 1, paramItem)
                        }

```

This way if you need a typed NULL you can pass it in a list as a typed parameter, otherwise it uses the class of the `paramItem` to determine the SQL type, like it does normally.

I pushed the fix to the repo and updating maven now. It may take a bit for central to show the new release 0.5.2.

**Comment by vsch** [comment]
@comm1x, I made a release although Travis build is failing. I did update Kotlin to latest 1.3.71 so will need to figure out why travis build fails.

I does build locally with maven so you should be able to use it.

**Comment by vsch** [comment]
@comm1x, the released build is good. I had travis language set to `kotlin` instead of `java`. It passes now.

**Comment by comm1x** [comment]
@vsch fixed in 0.5.2! Now it works, thanks a lot! ❤️
